src/builtins/coefficient/README.md

```diff
@@ -0,0 +1,92 @@
+# Coefficient
+
+`Coefficient` is a generic plugin for multiplying a value in an `input` array by a given coefficient.
+
+You provide the name of the value you want to multiply, a coefficient value, and a name to use to append the product to the output array.
+
+For example, you could multiply `cpu/energy` by 10 and name the result `energy-product`. `energy-product` would then be added to every observation in your input array as the product of `cpu/energy` and 10.
+
+## Parameters
+
+### Plugin global config
+
+Three parameters are required in global config: `input-parameter`, `coefficient` and `output-parameter`.
+
+- `input-parameter`: a string matching an existing key in the `inputs` array
+- `coefficient`: the value to multiply `input-parameter` by.
+- `output-parameter`: a string defining the name to use to add the product of the input parameters to the output array.
+
+### Inputs
+
+All of `input-parameters` must be available in the input array.
+
+## Returns
+
+- `output-parameter`: the product of all `input-parameters` with the parameter name defined by `output-parameter` in global config.
+
+## Calculation
+
+```pseudocode
+output = input * coefficient
+```
+
+## Implementation
+
+To run the plugin, you must first create an instance of `Coefficient`. Then, you can call `execute()`.
+
+```typescript
+const config = {
+  'input-parameter': 'carbon',
+  coefficient: 10,
+  'output-parameter': 'carbon-product',
+};
+
+const coeff = Coefficient(config);
+const result = coeff.execute([
```

Comment:
@jmcook1186 @manushak do we need this implementation part for builtin plugins?

Reply:
I think it's still useful for people to understand how the plugin works